openfe/setup/atommapping.py

```diff
@@ -19,14 +24,99 @@ class AtomMapping:
       If this atom has no corresponding atom, None is returned.
 
     """
+
     mol1: Molecule
     mol2: Molecule
-    mol1_to_mol2: Dict[int, int]
+    mol1_to_mol2: AtomMap
 
     def __hash__(self):
-        return hash((hash(self.mol1), hash(self.mol2),
-                     tuple(self.mol1_to_mol2.items())))
+        return hash(
+            (hash(self.mol1), hash(self.mol2), tuple(self.mol1_to_mol2.items()))
+        )
 
     @classmethod
     def from_perses(cls, perses_mapping):
         raise NotImplementedError()
+
+    def _match_elements(self, idx: int):
+        elem_mol1 = self.mol1.GetAtomWithIdx(idx).GetAtomicNum()
+        elem_mol2 = self.mol2.GetAtomWithIdx(self.mol1_to_mol2[idx]).GetAtomicNum()
+        if elem_mol1 == elem_mol2:
+            return True
+        else:
+            return False
+
+    def _get_unique_bonds_and_atoms(self, mapping: AtomMap):
+
+        uniques = {
+            "atoms": [],  # atoms which fully don't exist in mol2
+            "elements": [],  # atoms which exist but change elements in mol2
+            "bonds": [],  # bonds involving either unique atoms or elements
```

Comment:
As commented below, consider using sets here. These should have no duplicates and order doesn't matter, right? If so, then `set` is the appropriate container.